src/main/java/net/fabricmc/loader/entrypoint/minecraft/EntrypointPatchBranding189.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.entrypoint.minecraft;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.entrypoint.EntrypointPatch;
+import net.fabricmc.loader.entrypoint.EntrypointTransformer;
+import net.fabricmc.loader.launch.common.FabricLauncher;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+
+import java.io.IOException;
+import java.util.ListIterator;
+import java.util.function.Consumer;
+
+public class EntrypointPatchBranding189 extends EntrypointPatch {
+
+	public EntrypointPatchBranding189(EntrypointTransformer transformer) {
+		super(transformer);
+	}
+
+	@Override
+	public void process(FabricLauncher launcher, Consumer<ClassNode> classEmitter) {
+		if(FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT){
+			loadClass(launcher, "net/minecraft/client/MinecraftClient").methods.forEach(m -> {
```

Comment:
What is the obf name of Minecraft class in 1.8.9? Use the obf name as I don't think such a deobf class exists in 1.8.9 binary

Reply:
that is the obf name

Reply:
u have to choose 1 file to not obfuscate with proguard, and mojang usually goes with the main class, you'll see it in newer versions too

Reply:
isn't it `net/minecraft/client/Main` for client tho

Reply:
@hYdos I just looked at 1.8.9 jar, the clas should be `ave`

Reply:
oh really? thats weird i swear thats what it is in this version

Reply:
Didn't you test this? Also, I wouldn't rely on obfuscated names like `ave`, should inspect the class first.